app/models/grda_warehouse/auth_policies/policy_provider.rb

```diff
@@ -0,0 +1,99 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'memery'
+
+class GrdaWarehouse::AuthPolicies::PolicyProvider
+  include Memery
+  attr_reader :user
+
+  # TODO: START_ACL remove after ACL migration is complete
+  ###
+  # The legacy permission policy only checks the permissions on a user's roles and assumes the loaded record is visible
+  # to the user already. This is a dangerous assumption so it's required to confirm the record is authorized by
+  # setting legacy_implicitly_assume_authorized_access to true.
+  # Note this is only relevant for users where using_acls is false
+  ###
+  attr_accessor :legacy_implicitly_assume_authorized_access
+  # END_ACL
+
+  def initialize(user)
+    @user = user
+  end
+
+  def for_client(client_or_id)
+    client_id = id_from_arg(client_or_id, GrdaWarehouse::Hud::Client)
+    if user.using_acls?
+      client_project_policy(client_id)
+    else
+      # TODO: START_ACL remove after ACL migration is complete
+      legacy_user_role_policy
+      # END_ACL
+    end
+  end
+
+  def for_patient(patient)
+    for_client(patient.client_id)
+  end
+
+  memoize def for_project(project_or_id)
+    project_id = id_from_arg(project_or_id, GrdaWarehouse::Hud::Project)
+    if user.using_acls?
+      GrdaWarehouse::AuthPolicies::ProjectPolicy.new(user: user, project_id: project_id)
+    else
+      # TODO: START_ACL remove after ACL migration is complete
+      legacy_user_role_policy
+      # END_ACL
+    end
+  end
+
+  protected
+
+  def handle_legacy_unauthorized
+    raise 'legacy authorization not performed'
+  end
+
+  memoize def client_project_policy(client_id)
+    project_policies = visible_client_project_ids(client_id).map do |project_id|
+      for_project(project_id)
+    end
+    GrdaWarehouse::AuthPolicies::AnyPolicy.new(policies: project_policies)
+  end
+
+  # TODO: START_ACL remove after ACL migration is complete
+  memoize def legacy_user_role_policy
+    handle_legacy_unauthorized unless legacy_implicitly_assume_authorized_access
+    GrdaWarehouse::AuthPolicies::LegacyUserRolePolicy.new(user: user)
+  end
+  # END_ACL
+
+  # Needs review
```

Comment:
I believe this is correct because it enforces `can_view_clients` which in the ACL world is needed for both viewing a client and searching. (see note on `can_search_own_clients` in the role class.)

Reply:
I wonder, maybe it's better to just query the projects for all the client's enrollments since the checks are performed in ProjectPolicy?
```
  memoize def client_project_policy(client_id)
    c_t = GrdaWarehouse::Hud::Client.arel_table
    client_project_scope = GrdaWarehouse::Hud::Project.joins(:clients).where(c_t[:id].eq(client_id))
    project_policies = client_project_scope.pluck(:id).map do |project_id|
      for_project(project_id)
    end
    GrdaWarehouse::AuthPolicies::AnyPolicy.new(policies: project_policies)
  end
```